### PR TITLE
Use async nats in rpc server and client

### PR DESCRIPTION
**Pull request by victor-carvalho**

### CODE REVIEW

**Comment by rodopoulos** [comment]
The version bump is missing. Apart from that, LGTM. I'd wait for @leohahn review, as well.

File: src/pitaya_etcd_nats_cluster/src/rpc_client.rs [hunk 1]
```diff
@@ -109,30 +119,19 @@ impl RpcClient for NatsRpcClient {
         let request_timeout = self.settings.request_timeout;
 
         // We do a spawn_blocking here, since it otherwise will block the executor thread.
```
Comment (leohahn) [inline src/pitaya_etcd_nats_cluster/src/rpc_client.rs]: This comment seems outdated.

File: src/pitaya_etcd_nats_cluster/src/rpc_server.rs [hunk 2]
```diff
@@ -13,9 +15,24 @@ use tokio::sync::{mpsc, oneshot, RwLock};
...
 const RPCS_IN_FLIGHT_METRIC: &str = "rpcs_in_flight";
 
+struct RpcServerState {
+    connection: asynk::Connection,
+    close_sender: oneshot::Sender<()>,
+}
+
+impl RpcServerState {
+    async fn close(self) -> Result<(), Error> {
+        // TODO(victor-carvalho) we have to wait on connection close
```
Comment (leohahn) [inline src/pitaya_etcd_nats_cluster/src/rpc_server.rs]: This todo seems outdated.

File: src/pitaya_etcd_nats_cluster/src/rpc_server.rs [hunk 3]
```diff
@@ -13,9 +15,24 @@ use tokio::sync::{mpsc, oneshot, RwLock};
...
+        let _ = self.close_sender.send(());
+        self.connection.close().await.map_err(Error::Nats)
+    }
+}
+
+type NatsRpcServerState = Arc<RwLock<Option<RpcServerState>>>;
```
Comment (leohahn) [inline src/pitaya_etcd_nats_cluster/src/rpc_server.rs:28]: Since the type is private, you can remove the Nats prefix.

**Comment by leohahn** [comment]
Apart from the comments, LGTM. You also need to rebase the branch.

File: src/pitaya_etcd_nats_cluster/src/rpc_server.rs (hunk 3 above)
Comment (victor-carvalho) [reply src/pitaya_etcd_nats_cluster/src/rpc_server.rs:28]: I only added the Nats because RpcServerState already exists

**Pull request merged**